Makefile

```diff
@@ -561,3 +567,18 @@ cert-manager: check-local-context kubectl ## install cert-manager to cluster
 python-init:
 	python3 -m venv tests/e2e/.venv
 	tests/e2e/.venv/bin/pip install -r requirements.txt
+
+.PHONY: etcd
+etcd: $(ETCD)
+$(ETCD): $(LOCALBIN)
+	@test -s $(LOCALBIN)/etcd || { \
+		if [ "$(GOOS)" = "darwin" ]; then \
+			curl -sfSLo etcd-temp https://github.com/etcd-io/etcd/releases/download/$(ETCD_VERSION)/etcd-$(ETCD_VERSION)-$(GOOS)-$(GOARCH).zip && \
+			unzip -j etcd-temp "*/etcd" "*/etcdctl" -d $(LOCALBIN) && \
+			rm etcd-temp; \
+		else \
+			curl -sfSLo etcd-temp https://github.com/etcd-io/etcd/releases/download/$(ETCD_VERSION)/etcd-$(ETCD_VERSION)-$(GOOS)-$(GOARCH).tar.gz && \
+			tar -xvf etcd-temp -C $(LOCALBIN) --strip-components=1 && \
+			rm etcd-temp; \
+		fi \
+	}
```

Comment:
Ouch. Thinking of ways to make this a little cleaner, maybe:

1. Move the `-o` in `-o etcd-temp` into its own argument, so that it's more clearly deliniated
2. Store the temp file in `$(LOCALBIN)` instead of in the current directory?

Reply:
Sounds good